stylus-sdk/src/host/mod.rs

```diff
@@ -0,0 +1,364 @@
+// Copyright 2024-2025, Offchain Labs, Inc.
+// For licensing, see https://github.com/OffchainLabs/stylus-sdk-rs/blob/main/licenses/COPYRIGHT.md
+
+//! Defines host environment methods Stylus SDK contracts have access to.
+use alloc::vec::Vec;
+use alloy_primitives::{Address, B256, U256};
+
+use crate::call::{MutatingCallContext, StaticCallContext};
+
+/// The host trait defines methods a Stylus contract can use to interact
+/// with a host environment, such as the EVM. It is a composition
+/// of traits with different access to host values and modifications.
+/// Stylus contracts in the SDK parametrized by a host trait allow for safe access
+/// to hostios without the need for global invocations. The host trait may be implemented
+/// by test frameworks as an easier way of mocking hostio invocations for testing
+/// Stylus contracts.
+pub trait Host:
+    CryptographyAccess
+    + CalldataAccess
+    + DeploymentAccess
+    + StorageAccess
+    + CallAccess
+    + BlockAccess
+    + ChainAccess
+    + AccountAccess
+    + MemoryAccess
+    + MessageAccess
+    + MeteringAccess
+{
+}
+
+/// Defines a trait that allows a Stylus contract to access its host safely.
+pub trait HostAccess<H: Host> {
+    /// Provides access to the parametrized host value of a contract.
+    fn get_host(&self) -> &H;
```

Comment:
Could this return an instance of the host instead of a reference? We had this conversation before, but I lost the thread.

Reply:
I will edit, and see if we run into any issues during implementation, as it's hard to tell at the moment